text/0000-wheel-platform-support.md

```diff
@@ -0,0 +1,217 @@
+
+# Guide-level explanation
+
+When installing gems with native extensions, RubyGems will now automatically select the most appropriate pre-compiled binary based on your exact Ruby version and platform:
+
+```bash
+$ gem install nokogiri
+# On Ruby 3.3, x86_64 Linux with glibc 2.28:
+# Installs nokogiri-1.16.0-whl-rb33-manylinux_2_28_x86_64.gem (pre-compiled)
+
+# On Ruby 3.2, same platform:
+# Falls back to nokogiri-1.16.0.gem (compiles from source)
+```
+
+Gem authors can build platform-specific gems:
+
+```sh
+gem build --platform "whl-rb33-x86_64_linux"  # Specific to Ruby 3.3 on Linux x86_64
+```
+
+
+The wheel platform format follows the pattern `whl-{abi_tags}-{platform_tags}`:
+
+- `abi_tags`: Ruby ABI version (e.g., `rb33` for Ruby 3.3, `any` for pure Ruby, `cr3` for any CRuby with major version 3)
+- `platform_tags`: Platform specification (e.g., `x86_64_linux`, `manylinux_2_28_x86_64`, `arm64_darwin_23`)
+
+Multiple tags can be combined with dots to express compatibility with multiple versions:
+
+```
+whl-rb33.rb32-x86_64_linux  # Works with Ruby 3.3 or 3.2 on Linux x86_64
+```
+
+# Reference-level explanation
+
+The implementation adds four new classes to `lib/rubygems/platform/`:
+
+**`Gem::Platform::Wheel`** - Parses wheel-format platform strings (`whl-{abi}-{platform}`)
+- Handles multi-tag specifications (e.g., `rb33.rb32` for multiple Ruby versions)
+- Provides matching logic against Ruby environments
+- Expands tag combinations for compatibility checking
+
+**`Gem::Platform::Specific`** - Represents the current Ruby environment with precision
+- Detects Ruby engine, version, and ABI version
+- Identifies Linux libc type and version through ELF analysis
+- Generates appropriate compatible wheel tags for the environment
+
+**`Gem::Platform::ELFFile`** - Analyzes ELF binaries on Linux
+- Reads interpreter path to identify musl vs glibc
+- Parses dynamic symbols to determine minimum libc version
+- Provides platform detection without external dependencies
+
+**`Gem::Platform::Manylinux`/`Gem::Platform::Musllinux`** - Linux compatibility standards
+- Implements PEP 600 (manylinux) and PEP 656 (musllinux) specifications
+- Maps glibc/musl versions to compatibility tags
+- Allows for compact representation of cross-distribution compatible binaries
+
+## Platform Resolution Changes
+
+The platform matching system now uses `Gem::Platform::Specific.local` instead of `Gem::Platform.local` for more precise matching. The `sort_priority` method in `Gem::Platform` now assigns:
+
+1. Ruby platform: priority -1 (highest, for pure Ruby gems)
```

Comment:
Why `-1` is the highest priority? It seems that it should be 3 or larger.

Reply:
I agree. This doesn't provide a new value.

Reply:
The proposal has been revised to show the priority order used to choose a platform when installing gems.